### PR TITLE
Add a JSON formatter

### PR DESCRIPTION
**Pull request by Arcanemagus**
Add a JSON formatter option, allowing easier use in tools that have native modes of parsing this output.

### CODE REVIEW

**Comment by steelbrain** [comment]
:+1:

**Comment by SimenB** [comment]
Publish it standalone on npm instead, then we can use it with the api & grunt (soon gulp as well hopefully) at least

**Comment by steelbrain** [comment]
@SimenB we have a working fork here, You can use it until this is merged, You'll have to point npm at git+https though. https://github.com/AtomLinter/csslint

**Comment by Arcanemagus** [comment]
Any chance that this could be merged or at least reviewed? I would absolutely love to get rid of [AtomLinter/csslint](https://github.com/AtomLinter/csslint) and switch [AtomLinter/linter-csslint](https://github.com/AtomLinter/linter-csslint) back to this official repository now that it seems to be active again, however without this PR I can't do that yet.

**Comment by Arcanemagus** [comment]
Are there any changes that need to be made before this can be merged in @frvge?

**Comment by frvge** [comment]
@Arcanemagus , I'm currently doing mostly quick triages to try to get the project alive again. At first sight your PR looks good. I'll add triage to get some other people's opinion.

**Comment by Arcanemagus** [comment]
*bump*

File: tests/formatters/json.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        name: "JSON formatter",
+
+        "File with no problems should say so": function() {
+            var result = { messages: [], stats: [] },
+                actual = CSSLint.getFormatter("json").formatResults(result, "path/to/FILE", {fullPath: "/absolute/path/to/FILE"});
+            Assert.areEqual("{\"messages\":[],\"stats\":[]}", actual);
+        },
+
+        "Should have no output when quiet option is specified and no errors": function() {
```
Comment (ideadapt) [inline tests/formatters/json.js]: Shouldn't you define errors & warnings in your result object to test if quiet has an effect? 
What happens if I set `quiet: "false"` ?

File: src/formatters/json.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    },
+
+    /**
+     * Given CSS Lint results for a file, return output for this format.
+     * @param results {Object} with error and warning messages
+     * @return {String} output for results
+     */
+    formatResults: function(results, filename, options) {
+        "use strict";
+        return options.quiet ? "" : JSON.stringify(results);
```
Comment (Arcanemagus) [inline src/formatters/json.js]: Can a `JSON` object be relied upon to exist in all the environments `csslint` is supported under?

File: tests/formatters/json.js (hunk 1 above)
Comment (Arcanemagus) [inline tests/formatters/json.js]: I'm not sure I understand what you are asking here.

The way `JSON.stringify` works means that the result object being passed in there would still show `{"messages":[],"stats":[]}` if quiet is not true.

Formatters have nothing to do with parsing the options, if `csslint` isn't sending in a properly valued `quiet` option that's a bug in the rest of `csslint`, right?

Comment (Arcanemagus) [inline tests/formatters/json.js]: Wow, I just realized what you mean, and that's a silly bug, fixing.

File: src/formatters/json.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    /**
+     * Given CSS Lint results for a file, return output for this format.
+     * @param results {Object} with error and warning messages
+     * @param filename {String} relative file path (Unused)
+     * @return {String} output for results
+     */
+    formatResults: function(results, filename, options) {
+        "use strict";
+        return options.quiet && results.messages.length < 1 ? "" : JSON.stringify(results);
```
Comment (Arcanemagus) [inline src/formatters/json.js]: `jshint` complains about `JSON` being unavailable, and it looks like rightfully so as one of the supported platforms is WSH, which may or may not have this defined, depending on the version being run.

I'm not sure how to solve this though, any pointers?

**Comment by Arcanemagus** [comment]
Added a commit specifying `JSON` as a global variable. This is fine in Node.js and Rhino, but can potentially cause issues in WSH as it is only available from JScript 5.8 onwards. It seems actually _using_ JScript 5.8 though can be [quite the pain](http://stackoverflow.com/questions/19445189/cscript-jscript-json).

As the WSH interface is currently not supported, whether this is an issue or not I'll leave to you guys.

**Comment by Arcanemagus** [comment]
Also as a note: None of the other formatters actually test if they properly output messages if there are ones to display and `--quiet` has been specified... of the ones that even support/test that functionality in the first place!

**Comment by Arcanemagus** [comment]
Any further changes? Comments?

**Comment by frvge** [comment]
@XhmikosR I re-ran the tests to verify an issue with another PR and it looks like travis is having issues. Do you know anything about that?

**Comment by XhmikosR** [comment]
@frvge: it's a known issue with node.js 5.7.0. 5.7.1 should be out today or so, so it will be fixed.

**Comment by Arcanemagus** [comment]
Btw, working on incorporating proper multiple file support here.

**Comment by Arcanemagus** [comment]
There, should properly handle multiple files now. If there are any changes required please let me know. (Should I be updating the `dist` files?)

**Comment by Arcanemagus** [comment]
Any further changes that need to be made?

**Comment by Arcanemagus** [comment]
Are there any issues still needing to be resolved here?
Is this blocked due to the specific WSH version requirement?

If this PR and #605 weren't forcing [linter-csslint](https://github.com/AtomLinter/linter-csslint) to use a custom (unmaintained!) fork of `csslint` I would have long ago abandoned this...

**Comment by Arcanemagus** [comment]
Bump...

**Comment by frvge** [comment]
@XhmikosR ? I think it looks good.

**Pull request merged**